### PR TITLE
py-nanonbind: add new versions

### PR DESCRIPTION
**Pull request by garth-wells**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-nanobind/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanobind/package.py]: These commit hashes don't seem right, I'm seeing b0e24d5b0ab0d518317d6b263a257ae72d4d29a2 for v1.5.2

Comment (garth-wells) [reply var/spack/repos/builtin/packages/py-nanobind/package.py]: Fixed: I used the file checksums :).

**Review by adamjstewart — DISMISSED** [review]
Will give @ma595 a chance to review before merging, ping me in a week if you haven't heard anything

**Comment by adamjstewart** [comment]
@spackbot fix style

**Review by adamjstewart — APPROVED** [review]

**Review by bernhardkaindl — APPROVED** [review]
@ma595 4 days passed without a review from you. After 7 days, we'd merge it.

**Review by ma595 — APPROVED** [review]

**Review by ma595 — APPROVED** [review]

**Comment by garth-wells** [comment]
@adamjstewart: @ma595 has approved this PR.

**Pull request merged**
